recipes/vtk-m/meta.yaml

```
@@ -0,0 +1,91 @@
+{% set name = "vtk-m" %}
+{% set version = "2.0.0" %}
+{% set soversion = version.split('.')[0] + "." + version.split('.')[1] %}
+{% set common_flags = "-DCMAKE_BUILD_TYPE=Release -DVTKm_ENABLE_EXAMPLES=ON -DVTKm_INSTALL_EXAMPLES=OFF" %}
+{% set vtkm_modules = [
+       'cont',
+       'filter_clean_grid',
+       'filter_connected_components',
+       'filter_contour',
+       'filter_core',
+       'filter_density_estimate',
+       'filter_entity_extraction',
+       'filter_field_conversion',
+       'filter_field_transform',
+       'filter_flow',
+       'filter_geometry_refinement',
+       'filter_image_processing',
+       'filter_mesh_info',
+       'filter_multi_block',
+       'filter_resampling',
+       'filter_scalar_topology',
+       'filter_vector_analysis',
+       'filter_zfp',
+       'io',
+       'rendering',
+       'source',
+       'worklet',
+] %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://gitlab.kitware.com/vtk/vtk-m/-/archive/v{{ version }}/{{ name }}-v{{ version }}.tar.gz
+  sha256: 32643cf3564fa77f8e2a2a5456a574b6b2355bb68918eb62ccde493993ade1a3
+
+build:
+  script:
+    - cmake -GNinja -S $SRC_DIR  -B build -DCMAKE_INSTALL_PREFIX=$PREFIX {{ common_flags }}  # [unix]
+    - cmake -GNinja -S %SRC_DIR% -B build -DCMAKE_INSTALL_PREFIX=%PREFIX% {{ common_flags }}  # [win]
+    - cmake --build build -j
+    - cmake --install build
+    - VTKm_DIR=$PREFIX cmake -GNinja -S $SRC_DIR/examples/smoke_test/ -B smoke_test_build  # [unix]
+    - cmake --build smoke_test_build  # [unix]
+    - ./smoke_test_build/smoke_test  # [unix]
+  number: 0
+  run_exports:
+    - {{ pin_subpackage(name|lower, max_pin=soversion) }}
```

```suggestion
    - {{ pin_subpackage(name|lower, max_pin='x.x') }}
```
Max pin should literally by a string of multiple `x` separated by `.` i.e. if the ABI is stable within minor API versions only, then the max pin would be `'x.x'`.

Literal version numbers use a different parameter.

https://docs.conda.io/projects/conda-build/en/stable/resources/variants.html#pinning-at-the-recipe-level